ui/tests/e2e/extensions.spec.js

```diff
@@ -0,0 +1,133 @@
+    await expect(enableButton).toBeVisible();
+    await expect(enableButton).toBeDisabled();
+
+    // Verify snapshot logo using data-testid
+    await expect(page.getByTestId('kanvas-snapshot-image')).toBeVisible();
+  });
+
+  test('Verify Performance Analysis Details', async ({ page }) => {
+    // Verify heading using data-testid
+    await expect(page.getByTestId('performance-analysis-heading')).toBeVisible();
+
+    // Verify enable button using data-testid
+    const performanceEnableButton = page.getByTestId('performance-analysis-enable-btn');
+    await expect(performanceEnableButton).toBeVisible();
+    await expect(performanceEnableButton).toBeDisabled();
+  });
+
+  test('Verify Kanvas Details', async ({ page, context }) => {
+    // Verify Kanvas heading using data-testid
+    await expect(page.getByTestId('kanvas-signup-heading')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('kanvas-signup-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL('https://docs.layer5.io/kanvas/');
+    await newPage.close();
+  });
+
+  test('Verify Meshery Docker Extension Details', async ({ page, context }) => {
+    // Verify Docker Extension heading using data-testid
+    await expect(page.getByTestId('docker-extension-heading')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('docker-extension-download-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL(
+      'https://hub.docker.com/extensions/meshery/docker-extension-meshery',
+    );
+    await newPage.close();
+  });
+
+  test('Verify Meshery Design Embed Details', async ({ page, context }) => {
+    await expect(page.getByTestId('design-embed-learn-more-btn')).toBeVisible();
+
+    const [newPage] = await Promise.all([
+      context.waitForEvent('page'),
+      page.getByTestId('design-embed-learn-more-btn').click(),
+    ]);
+
+    await expect(newPage).toHaveURL('https://docs.layer5.io/kanvas/designer/embedding-designs/');
+    await newPage.close();
+  });
+
+  test('Verify Meshery Catalog Section Details', async ({ page, context }) => {
+    // Verify Catalog section heading using data-testid
+    await expect(page.getByTestId('catalog-section-heading')).toBeVisible();
+
+    // Verify catalog toggle switch using data-testid
+    await expect(page.getByTestId('catalog-toggle-switch')).toBeVisible();
+
+    // First verify the toggle button functionality
+    const toggleButton = page
+      .locator(
+        'div:has-text("Explore the Meshery Catalog") >> .MuiSwitch-root input[type="checkbox"]',
+      )
+      .nth(0);
+    await expect(toggleButton).toBeVisible();
+
+    // Click toggle Button and Verify the State changes
+    const initialState = await toggleButton.isChecked();
+    await toggleButton.click();
+    await expect(toggleButton).toBeChecked(!initialState);
```

Comment:
When we interact w/ a button the action implicitly verifies several items and visibility is one of those. So we can remove this     await expect(toggleButton).toBeVisible(); 

Also can we just test if it can be clicked? Not sure if the toggle button logic is complex here to warrent checking state changed? 

https://playwright.dev/docs/input#mouse-click